src/Stratis.Bitcoin.Features.PoA.IntegrationTests/Tools/FullNodePoATestBuilderExtension.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Stratis.Bitcoin.Builder;
+using Stratis.Bitcoin.Builder.Feature;
+
+namespace Stratis.Bitcoin.Features.PoA.IntegrationTests.Tools
+{
+    public static class FullNodePoATestBuilderExtension
+    {
+        public static IFullNodeBuilder AddFastMiningCapability(this IFullNodeBuilder fullNodeBuilder)
+        {
+            fullNodeBuilder.ConfigureFeature(features =>
+            {
+                foreach (IFeatureRegistration feature in features.FeatureRegistrations)
+                {
+                    feature.FeatureServices(services =>
+                    {
+                        ServiceDescriptor defaultProivider = services.FirstOrDefault(x => x.ServiceType == typeof(PoAMiner));
+
+                        services.Remove(defaultProivider);
```

Comment:
Not sure if needed. It will require even more code and will be harder to read- https://stackoverflow.com/questions/43590769/replace-service-registration-in-asp-net-core-built-in-di-container

also we use remove and add in other places

Reply:
https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Stratis.Bitcoin.Features.SmartContracts/SmartContractFeature.cs#L122